src/Core/Challenge/Dns/OvhSolver.php

```diff
@@ -0,0 +1,243 @@
+use Psr\Log\NullLogger;
+use Webmozart\Assert\Assert;
+
+/**
+ * ACME DNS solver with automate configuration of a Gandi.Net.
+ *
+ * @author Alexander Obuhovich <aik.bold@gmail.com>
+ */
+class OvhSolver implements MultipleChallengesSolverInterface, ConfigurableServiceInterface
+{
+    use LoggerAwareTrait;
+    /**
+     * @var DnsDataExtractor
+     */
+    private $extractor;
+
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @var array
+     */
+    private $cacheZones;
+
+    /**
+     * @var string
+     */
+    private $appKey;
+
+    /**
+     * @var string
+     */
+    private $appSecret;
+
+    /**
+     * @var string
+     */
+    private $endPoint;
+
+    /**
+     * @var string
+     */
+    private $consumerKey;
+
+    /**
+     * @var string
+     */
+    private $domainDelegation;
+
+    /**
+     * @param DnsDataExtractor $extractor
+     * @param ClientInterface  $client
+     */
+    public function __construct(
+        DnsDataExtractor $extractor = null,
+        ClientInterface $client = null
+        ) {
+        $this->extractor = null === $extractor ? new DnsDataExtractor() : $extractor;
+        $this->client = null === $client ? new Client() : $client;
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * Configure the service with a set of configuration.
+     */
+    public function configure(array $config)
+    {
+        $this->appKey = $config['app_key'];
+        $this->appSecret = $config['app_secret'];
+        $this->endPoint = $config['end_point'];
+        $this->consumerKey = $config['consumer_key'];
+        if (isset($config['domain_delegation'])) {
+            $this->domainDelegation = $config['domain_delegation'];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(AuthorizationChallenge $authorizationChallenge)
+    {
+        return 'dns-01' === $authorizationChallenge->getType();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function solve(AuthorizationChallenge $authorizationChallenge)
+    {
+        return $this->solveAll([$authorizationChallenge]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function solveAll(array $authorizationChallenges)
+    {
+        Assert::allIsInstanceOf($authorizationChallenges, AuthorizationChallenge::class);
+
+        foreach ($authorizationChallenges as $authorizationChallenge) {
+            $this->logger->debug('Create TXT for '.$authorizationChallenge->getDomain());
+
+            $topLevelDomain = $this->getTopLevelDomain($authorizationChallenge->getDomain());
+            $recordName = $this->extractor->getRecordName($authorizationChallenge);
+            $recordValue = $this->extractor->getRecordValue($authorizationChallenge);
+
+            $subDomain = \str_replace('.'.$topLevelDomain.'.', '', $recordName);
+
+            $client = new Client(['http_errors' => false]);
+
+            $ovh = new Api($this->appKey,
```

Comment:
The coding style here seems weird: you can either put everything on the same line, or all arguments on a new line